OpenHPL/Waterway/Drafttube/Elbow.mo

```diff
@@ -0,0 +1,94 @@
+     parameter Modelica.SIunits.Length R_o= 7 "Radius of curvature through the center of the elbow" annotation (
+    Dialog(group = "Geometry"));
+  parameter Modelica.SIunits.Height p_eps = data.p_eps "Pipe roughness height" annotation (
+    Dialog(group = "Geometry"));
+  //// condition of steady state
+  parameter Boolean SteadyState = data.Steady "if true - starts from Steady State" annotation (
+    Dialog(group = "Initialization"));
+  //// staedy state value for flow rate
+  parameter Modelica.SIunits.VolumeFlowRate Vdot_0 = data.V_0 "Initial flow rate in the cone" annotation (
+    Dialog(group = "Initialization"));
+  //// possible parameters for temperature variation. Not finished...
+  //parameter Boolean TempUse = data.TempUse "If checked - the water temperature is not constant" annotation (Dialog(group = "Initialization"));
+  //parameter Modelica.SIunits.Temperature T_0 = data.T_0 "Initial water temperature in the pipe" annotation (Dialog(group = "Initialization", enable = TempUse));
+
+  // variables
+  Modelica.SIunits.Momentum M "Water momentum";
+  Modelica.SIunits.Mass m "water mass";
+  Modelica.SIunits.Velocity v "Water velocity";
+  Modelica.SIunits.Volume V "Volume of water";
+  Modelica.SIunits.Diameter D_e,D_  "Diameters";
+  Modelica.SIunits.Length L_CM,EI,EM,IM  "Lengths";
+  Modelica.SIunits.Area A_,A_i,A_o,A_e "Areas";
+  Modelica.SIunits.Force F,F_p,Fg_Vdot,F_f,F_D,F_d "Forces";
+  Modelica.SIunits.Pressure p_i,p_o,p_e "Pressure";
+  Real phi_dk,zeta_d,zeta_k,kappa  "Non-dimensional values";
+
+  Modelica.SIunits.VolumeFlowRate Vdot(start = Vdot_0) "Flow rate";
+
+  //// variables for temperature. Not in use for now...
+  //Real W_f, W_e;
+  //Modelica.SIunits.Temperature T( start = T_0);
+
+  // connector
+  extends OpenHPL.Interfaces.ContactPort;
+
+initial equation
+  if SteadyState == true then
+    der(M) = 0;
+  end if;
+equation
+  der(M) = F;
+
+  // Equations as in the paper (link later)
+  // Relating M with algebraic variables and Vdot
+  M = m*v;
+  m = data.rho*V; V = pi*L/12*(D_i^2+D_e^2+D_i*D_e); D_e=D_o;
+  v = Vdot/A_; A_= pi*D_^2/4;mdot=data.rho*Vdot;
+  if D_i<D_e then
+      L_CM = L/16*(3*D_i^2+2*D_i*D_e+D_e^2)/(D_i^2+D_i*D_e+D_e^2);
+      D_ = D_e-(D_e-D_i)*L_CM/L;
+    else
+      L_CM =L/16*(D_i^2+2*D_i*D_e+3*D_e^2)/(D_i^2+D_i*D_e+D_e^2);
+      D_ = D_i-(D_i-D_e)*L_CM/L;
+  end if;
+
+  // Relating F with algebraic variables and p_i and p_e
+  F = F_p-Fg_Vdot-F_f;
+  F_p = p_i*A_i-p_e*A_e; A_i = pi*D_i^2/4; A_e = pi*D_e^2/4; p_e=p_o;A_o=A_e;
+  Fg_Vdot = m*data.g*H_d/EI;
+  EI = (IM^2+EM^2+2*IM*EM*cos(Modelica.SIunits.Conversions.from_deg(delta)))^2;IM = L_CM; EM = L-L_CM;
+  F_f = F_D + F_d;
+  F_D = Functions.DarcyFriction.Friction(v, D_, L, data.rho, data.mu, p_eps);
+  F_d = 1/2*data.rho*v*abs(v)*A_*phi_dk;
+  phi_dk = zeta_d*zeta_k;
+  zeta_d = 1.1e-7*delta^3 + 8.46e-5*delta^2 + 0.018*delta-0.02;
+  zeta_k = 2.24*kappa^4 + 11.2*kappa^3 + 21.26*kappa^2 - 18.35*kappa+6.24;
```

Comment:
@pandeysudan27 The same again here, are these coefficients specific for one geometry or universal?

Reply:
These expressions are also calculated from the hydraulic resistance handbook based on ratios of inner curvature and the bend angle of the elbow.

Reply:
Again, maybe better to put the original calculation with the input parameters in here.